src/math/lp/int_solver.cpp

```diff
@@ -944,28 +949,34 @@ bool int_solver::shift_var(unsigned j, unsigned range) {
     if (column_is_int(j)) {
         if (!inf_l) {
             l = impq(ceil(l));
-            if (!m.is_one())
-                l = impq(m*ceil(l/m));
         }
         if (!inf_u) {
             u = impq(floor(u));
-            if (!m.is_one())
-                u = impq(m*floor(u/m));
         }
     }
     if (!inf_l && !inf_u && l >= u)
         return false;
     if (inf_u) {
         SASSERT(!inf_l);
-        impq delta   = impq(random() % (range + 1));
-        impq new_val = l + m*delta;
+        impq new_val = l + impq(random() % (range + 1));
```

Comment:
I think this can be simplifier:

```
    impq new_val;
    if (m.is_one()) {
         new_val = l + impq(random() % (range + 1));
    }
    else {
         new_val = get_value(j) + m*impq(random() % (range + 1));
     }